File: 6.DecimalToOctal/CPP/src/DecimalToOctal.cpp
#include "../include/DecimalToOctal.hpp"  // Include the header file for DecimalToOctal

// Function to convert decimal to octal
std::string DecimalToOctal::decimalToOctal(int decimal) {
    std::string octal = "";  // Initialize an empty string variable named octal to store the octal representation.

    // Convert decimal to octal
    while (decimal > 0) {
        int remainder = decimal % 8;  // Calculate the remainder when decimal is divided by 8
        octal = std::to_string(remainder) + octal;  // Convert the remainder to a string and prepend it to the octal string
        decimal /= 8;  // Update decimal by dividing it by 8.
    }

    return octal;  // Return the octal representation as the result of the DecimalToOctal::decimalToOctal function.
}

File: 6.DecimalToOctal/PYTHON/DecimalToOctal.py
# Python program to convert decimal to octal
# Visit https://venkys.io for more information

class DecimalToOctal:
    # Function to convert decimal to octal
    def decimal_to_octal(self, decimal):
        # Initialize an empty string to store the octal representation.
        octal = ""

        # Convert decimal to octal
        while decimal > 0:
            remainder = decimal % 8  # Calculate the remainder when decimal is divided by 8.
            octal = str(remainder) + octal  # Insert the remainder at the beginning of the octal string.
            decimal //= 8  # Update decimal by dividing it by 8.

        return octal  # Return the octal representation.

# Test the function
decimal = int(input("Enter a decimal number: "))
converter = DecimalToOctal()
octal = converter.decimal_to_octal(decimal)
print("Octal:", octal)
